### ISSUE TITLE
from_dataframe?

### ISSUE THREAD

**Issue by hainm** [ParmEd/ParmEd#270]
should have this method too?

```python
In [9]: p = pmd.load_file("./data/Tc5b.top")
In [10]: df = p.to_dataframe()
```

like
```python
df.charge.value = new_value
df.mass.values = new_mass
new_p = pmd.Structure.from_dataframe(df)
```

Anyway, this is only an idea flash in my mind.

**Comment by swails** [comment]
I hesitated making it an alternate constructor because `to_dataframe` only exports atomic properties (with some attached residue properties).  So it doesn't store any connectivity/topology information, which means you can't use just a `DataFrame` to serialize a `Structure`.

I would naively think `Structure.from_dataframe(struct.to_dataframe())` would return a copy (deep or shallow, doesn't matter) of `struct`.  But in general, it won't even come close.

Another idea is `load_dataframe`, which can override atom properties from a `DataFrame` in an existing `Structure` instance.  So you could do something like:

```Python
import pandas as pd
df = struct.to_dataframe()
# Zero-out charge on all atoms that have a charge with a magnitude greater than 0.1 electrons
df.loc[df.charge.abs() > 0.1, 'charge'] = 0
# Write those charges back to struct
struct.load_dataframe(df) # This does not exist yet.
```

But I'm going to wait for someone to want to use this in their own work that will hopefully implement it themselves :).

**Comment by hainm** [comment]
Actually I meant  `struct.load_dataframe(df)` like you wrote. Just loading what `to_dataframe` exported.

`Structure.from_dataframe(df)` is meaningless (we do have `Structure`).

**Comment by swails** [comment]
Let's wait on this until someone has a compelling use-case.

**Comment by swails** [comment]
This is the appropriate implementation for #298

**Comment by swails** [comment]
Implemented by #380

**Comment by hainm** [comment]
+1
